src/launchpad/utils/android/bundletool.py

```diff
@@ -0,0 +1,183 @@
+"""Android bundletool wrapper utilities."""
+
+from __future__ import annotations
+
+import json
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Any, Sequence
+
+from pydantic import BaseModel, Field
+
+from ..file_utils import create_temp_directory
+from ..logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class BundletoolError(Exception):
+    """Raised when bundletool command fails."""
+
+    def __init__(self, message: str, returncode: int, stdout: str, stderr: str) -> None:
+        super().__init__(message)
+        self.returncode = returncode
+        self.stdout = stdout
+        self.stderr = stderr
+
+
+class Bundletool:
+    """Wrapper around Android's bundletool CLI utility."""
+
+    def __init__(self, bundletool_path: str | Path | None = None) -> None:
+        """Initialize bundletool wrapper.
+
+        Args:
+            bundletool_path: Optional path to bundletool executable file. If not provided,
+                will attempt to find bundletool in PATH.
+
+        Raises:
+            FileNotFoundError: If bundletool cannot be found at specified path or in PATH
+        """
+        if bundletool_path is None:
+            try:
+                result = subprocess.run(
+                    ["which", "bundletool"],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                )
+                bundletool_path = result.stdout.strip()
+            except subprocess.CalledProcessError as e:
+                raise FileNotFoundError("bundletool not found in PATH. Install with `brew install bundletool`") from e
```

Comment:
note: we'll have to revisit this once we get further along with the Docker stuff